pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -255,6 +256,16 @@ func (ku *Upgrader) upgradeMasterNodes(ctx context.Context) error {
 	}
 
 	upgradedMastersIndex := make(map[int]bool)
+	for i := 0; i < expectedMasterCount; i++ {
+		upgradedMastersIndex[i] = false
+		for _, vm := range *ku.ClusterTopology.MasterVMs {
+			vmName := *vm.Name
+			if strings.Contains(vmName, strconv.Itoa(i)) {
```

Comment:
I do not think this will work, the `ClusterID` portion of VM name can interfere with the `vmName, strconv.Itoa(i)` comparison. I would avoid comparisons, something around this:

```go
for _, vm := range *ku.ClusterTopology.UpgradedMasterVMs {
  ID := strings.TrimSuffix(ku.ClusterTopology.DataModel.Properties.GetMasterVMPrefix(), *vm.Name)
  upgradedMastersIndex[ID] = true
}
```